doc/_ext/collapsible.py

```diff
@@ -0,0 +1,56 @@
+import random
+import string
+
+from docutils import nodes
+from docutils.parsers.rst import Directive
+from docutils.parsers.rst.directives import unchanged
+
+
+class CollapsibleNode(nodes.General, nodes.Element):
+    pass
+
+
+def visit_collapsible_node(self, node):
```

Comment:
Function argument named "self" has special meaning in Python.
Please change it.

Reply:
These functions are to become class methods; so passing 'self' as a first argument is kind of a common practice as I can see in tutorials.